src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -0,0 +1,175 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This portion of the binder converts a <see cref="WithExpressionSyntax"/> into a <see cref="BoundExpression"/>.
+    /// </summary>
+    internal partial class Binder
+    {
+        private BoundExpression BindWithExpression(WithExpressionSyntax syntax, DiagnosticBag diagnostics)
+        {
+            // PROTOTYPE: this entire method is likely to change
+            var receiver = BindRValueWithoutTargetType(syntax.Receiver, diagnostics);
+            var receiverType = receiver.Type;
+
+            var lookupResult = LookupResult.GetInstance();
+            HashSet<DiagnosticInfo>? useSiteDiagnostics = null;
+            bool hasErrors = false;
+
+            if (receiverType is null || receiverType.IsVoidType())
+            {
+                diagnostics.Add(ErrorCode.ERR_InvalidWithReceiverType, syntax.Receiver.Location);
+                receiverType = CreateErrorType();
+            }
+
+            MethodSymbol? cloneMethod = null;
+            if (!receiverType.IsErrorType())
+            {
+                // PROTOTYPE: The receiver type must have a instance method called 'Clone' with no parameters
+                LookupMembersInType(
+                    lookupResult,
+                    receiverType,
+                    "Clone",
+                    arity: 0,
+                    ConsList<TypeSymbol>.Empty,
+                    LookupOptions.MustBeInstance | LookupOptions.MustBeInvocableIfMember,
+                    this,
+                    diagnose: false,
+                    ref useSiteDiagnostics);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    foreach (var symbol in lookupResult.Symbols)
+                    {
+                        if (symbol is MethodSymbol { ParameterCount: 0 } m)
+                        {
+                            cloneMethod = m;
+                            break;
```

Comment:
How does this handle multiple results? That can happen for example when `Clone` is overridden. This technique does not correspond to how invocations are handled, as there is no contract for the order of symbols in a lookup result.